src/data/battler-tags.ts

```diff
@@ -2125,6 +2130,41 @@ export class CritBoostTag extends BattlerTag {
   }
 }
 
+/**
+ * A stackable instance of crit boost granted by G-Max Chi Strike
+ */
+export class CritBoostStackableTag extends BattlerTag {
+  public stackCount: number = 0;
+
+  constructor() {
+    super(
+      BattlerTagType.CRIT_BOOST_STACKABLE,
+      BattlerTagLapseType.TURN_END,
+      1,
+      Moves.G_MAX_CHI_STRIKE,
+      undefined,
+      true,
+    );
+  }
+
+  override onAdd(pokemon: Pokemon): void {
+    this.stackCount += 1;
+    // This actually does not have any messages in the mainline games
+    globalScene.queueMessage(
+      i18next.t("battlerTags:critBoostOnAdd", { pokemonNameWithAffix: getPokemonNameWithAffix(pokemon) }),
+    );
+  }
+
+  override onOverlap(pokemon: Pokemon): void {
+    this.onAdd(pokemon);
+  }
```

Comment:
```suggestion
  }
  
  override lapse(pokemon: Pokemon, lapseType: BattlerTagLapseType): boolean {
    return lapseType !== BattlerTagLapseType.CUSTOM || super.lapse(pokemon, lapseType);
  }
```
(So that the crit boost doesn't disappear at the end of each turn)

Reply:
you can just change the lapse type in the constructor to `CUSTOM` instead

Reply:
The original `CritBoostTag` is also using `BattlerTagLapseType.TURN_END` for some reason.

Reply:
So previously crit boost was only lasting for a single turn? :pikastare:

Reply:
Well, it's still permanent because it has the workaround pigeon posted.

But both that and this tag can have their lapse type changed to simplify things :)